### DISCUSSION TITLE
How can I make constructor?

### DISCUSSION THREAD

**Discussion by aoi562227** [community/community#21460] [Other Feature Feedback, Questions, & Ideas]
<p><span class="hashtag">#include</span><br>
<span class="hashtag">#include</span><br>
<span class="hashtag">#include</span><br>
<span class="hashtag">#define</span> pi 3.14<br>
using namespace std;</p>
<p>class won{</p>
<pre><code>private:
	float answer;
	float radius;
	float nop;
public:
	won(float radius){
	this -&gt; radius = radius;
	}
	won(float radius, float nop){
		this -&gt; radius = radius;
		this -&gt; nop = nop;
	}
	
	float nul(float radius){//넓이 구하는 함수
		answer = pow(radius,2)*pi;
		return answer;	
	}
	
	float gu(float radius){//구의 넓이 구하는함수
		float circle = pow(radius,3); // r의 세제곱
		answer = pi*circle*4/3;
		return answer; 
	}
	
	float gidung(float nop){//원기둥 함수
		answer = nul(radius)*nop;
		return answer;
	}
	
	float bbul(){//원뿔 함수
		int dap = gidung(radius);
		answer = dap/3;
		return answer;
	}
</code></pre>
<p>};</p>
<p>int main(){<br>
float answer;<br>
int radius = 0;</p>
<p>while(1){<br>
cout &lt;&lt; "반지름 입력 : ";	<br>
cin &gt;&gt; radius;<br>
int nop;<br>
int select;</p>
<pre><code>system("cls");
cout &lt;&lt; "메뉴" &lt;&lt; endl &lt;&lt; endl;
cout &lt;&lt;"1. 원의 넓이 2. 구의 부피 3. 원기둥의 부피 4.원뿔의 부피" &lt;&lt; endl;

cin &gt;&gt; select;

switch(select){

		case 1://원의 넓이
		won a(radius);
		answer = a.nul(radius);
		cout &lt;&lt; "원의 넓이는 " &lt;&lt; answer &lt;&lt; "입니다";
	case 2://구의 부피
		won b(radius);
		answer = b.gu();
		cout &lt;&lt; "구의 부피는 " &lt;&lt; answer &lt;&lt; "입니다";
	case 3://원기둥의 부피
		cin &gt;&gt; nop;
		won c(radius, nop);
		answer = c.gidung();
		cout &lt;&lt; "원기둥의 부피는 " &lt;&lt; answer &lt;&lt; "입니다";
	case 4:;//원뿔의 부피
		cin &gt;&gt; nop;
		won d(radius, nop);
		answer = won.bbul();
		cout &lt;&lt; "원뿔의 부피는 " &lt;&lt; answer &lt;&lt; "입니다";
}
</code></pre>
<p>}<br>
return 0;<br>
}</p>
<p>It is not works please help me</p>

**Comment by fire-eggs** [comment, marked as answer]
<p>You’ll have to tell us a) what the goal is; b) how it “not works” before anyone can help.</p>
<p>Assuming this is C++, I see one issue: the lack of a “break;” statement at the end of each “case” block.</p>

**Comment by aoi562227** [comment]
<p>Thank you. I’m sorry for first question.</p>
